api/desecapi/serializers.py

```diff
@@ -145,7 +229,7 @@ def update(self, instance: RRset, validated_data):
 
         rrs_data = validated_data.pop('records', None)
         if rrs_data is not None:
-            instance.set_all_record_contents(set([rr_data['content'] for rr_data in rrs_data]))
+            instance.set_all_record_contents(set(rrs_data))
```

Comment:
Is the `set()` needed here?

Reply:
There are two questions involved:

1. What's our API's behavior when queried with creating an rrset with duplicate records?
2. If we gracefully handle duplicate records, where do we condense them.

I created a test that verifies the current behavior (gracefully accept) and annotated it with a TODO to discuss which behavior we want.

Reply:
As normalization is a bitch, we let pdns do the verification.